### ISSUE TITLE
attr_reader value cannot be accessed via super?

### ISSUE THREAD

**Issue by i110** [mruby/mruby#3966]
```ruby
class Foo
  attr_reader :val
  def initialize(val)
    @val = val
  end
end
class Bar < Foo
  def val
    super
  end
end
puts Foo.new('foo').val # OK
puts Bar.new('bar').val
```
The last line raises an exception:

```
trace (most recent call last):
        [0] hoge.rb:12
        [1] hoge.rb:9:in val
hoge.rb:9: Can't get cfunc env from non-cfunc proc. (TypeError)
```

Is this mruby limitation?

**Comment by shevegen** [comment]
Hmm. I tested the above with mruby 1.3.0 and get no error - it works fine in MRI and mruby.

**Comment by xor256** [comment]
in my machine working:
$./program.rb
foo
bar

```
ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]
and  mruby 1.2.0 (2015-11-17)
```

**Comment by i110** [comment]
@shevegen @xor256 Thank you for the information.
I did `git bisect` then found that https://github.com/mruby/mruby/commit/8f2c62407c0659d84126545e19505a851059e750 caused this problem. If you have a time please try 1.4.0 or later

**Comment by i110** [comment]
@matz Will this issue be fixed in 1.4.1?
